packages/walletconnect-solana/src/adapter.ts

```diff
@@ -115,16 +117,56 @@ export class WalletConnectWallet {
         }
     }
 
-    async signTransaction(transaction: Transaction): Promise<Transaction> {
+    async signTransaction<T extends Transaction | VersionedTransaction>(transaction: T): Promise<T> {
+        if (this._client && this._session) {
+            let rawTransaction: string;
+            let legacyTransaction: Transaction | undefined;
```

Comment:
let's call this `transactionToSign` since in one case this actually isn't the legacy transaction, its the VersionedTransaction

Reply:
@jnwng I checked the code. This variable is used always for legacyTransaction.

For versioned transaction it won't be initialized.

Reply:
you're right!